### DISCUSSION TITLE
Remove PHP_EOL or make it configurable

### DISCUSSION THREAD

**Discussion by c33s** [spatie/yaml-front-matter#31] [Q&A]
the issue is not that big but working on windows and always using linux line endings interfers with `PHP_EOL`. the issue is similar to https://dev.to/c33s/always-use--as-directory-seperator-in-php-43l7
simply use the linux/osx line  endings or make it somehow configurable (or even better read it from the markdown file and keep the line endings as they where in the markdown file). 
every resonable text editor is able to handle different line endings. if you edit files with regular notepad you probable won't use frontmatter at all.

have not tested it but can't it happen that if i am on windows and use a file with linux file endings the regex and the implode is not matching at all?

**Comment by sebastiandedeyne** [comment]
Would love to see an active Windows user pitch in here. Ideally this package is able to detect these kind of issues without additional configuration.

**Comment by c33s** [comment]
@sebastiandedeyne i am an active windows user and there are only two options, always use linux line endings which are understandable by linux, osx and many many editors on windows or add a configuration option for that. 

there is not really something to detect, if you go for detection the `PHP_EOL` is quite fine but in my experience the autodetection is the problem. because the generated code differs on the different platforms which is quite bad if you work with git diffs (we work with check out as is and commit as linux but this can still lead to huge diffs in the development which are only that big because of the different line endings)

so how can i help? my solution would be to hardcode linux line endings -> done. or as i wrote above make a simple variable or constructor argument where the user can define it. so if someone has a full stack windows chain then the person configures your package with windows line endings (even for people who work on linux) but in most cases i think its safe to stick with linux line endings (should be default - what i say as windows user :))

**Comment by sebastiandedeyne** [comment]
Sounds good!

How about we add a `bool $useCrlf = false` argument (does that sound good?) to `parse` and `parseFile`?

https://github.com/spatie/yaml-front-matter/blob/d4c46ba9671987bd7f839a44a01e5d54dfe1bd3c/src/YamlFrontMatter.php#L9

Could you send a PR?
